### PR TITLE
ARM VFP/NEON support

### PR DESCRIPTION
**Pull request by Deryugin**
Add support for VFP/NEON instructions. Currently it works only with NEON, but VFP is subset of NEON, so it's possible to cut it if necessary in future.

### CODE REVIEW

File: templates/arm/iwave_imx6/mods.config [hunk 1]
```diff
@@ -12,6 +12,8 @@ configuration conf {
 	include embox.arch.arm.vfork
 	@Runlevel(0) include embox.driver.cache.pl310(base_addr=0x00A02000)
 
+	include embox.arch.arm.fpu.vfp(log_level=4)
```
Comment (anton-bondarev) [inline templates/arm/iwave_imx6/mods.config]: log_level=4?

File: src/arch/arm/fpu/vfp.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#ifndef ARM_FPU_VFP_H_
+#define ARM_FPU_VFP_H_
+
+#define FPU_DATA_LEN 65 /* One word for coprocessor state
+			   and 64 words for VPF registers */
+
+#ifndef __ASSEMBLER__
+#include <string.h>
+static inline void arm_fpu_context_init(void *opaque) {
+	memset(opaque, 0, sizeof(uint32_t) * FPU_DATA_LEN);
```
Comment (anton-bondarev) [inline src/arch/arm/fpu/vfp.h]: It's really  necessary ?

File: templates/arm/iwave_imx6/mods.config (hunk 1 above)
Comment (Deryugin) [reply templates/arm/iwave_imx6/mods.config]: Fixed

File: src/arch/arm/fpu/vfp.h (hunk 2 above)
Comment (Deryugin) [reply src/arch/arm/fpu/vfp.h]: Well, probably this can be removed, but I think it's ok to make sure there's no garbage in this memory.

**Pull request merged**